parquet-avro/src/main/java/org/apache/parquet/avro/AvroWriteSupport.java

```diff
@@ -542,26 +543,24 @@ protected void writeDoubleArray(double[] array) {
     @Override
     public void writeCollection(GroupType schema, Schema avroSchema,
                                 Collection<?> array) {
-      if (array.size() > 0) {
+      if (!array.isEmpty()) {
         recordConsumer.startField(OLD_LIST_REPEATED_NAME, 0);
         try {
           for (Object elt : array) {
             writeValue(schema.getType(0), avroSchema.getElementType(), elt);
           }
         } catch (NullPointerException e) {
           // find the null element and throw a better error message
-          int i = 0;
-          for (Object elt : array) {
-            if (elt == null) {
-              throw new NullPointerException(
-                  "Array contains a null element at " + i + "\n" +
-                  "Set parquet.avro.write-old-list-structure=false to turn " +
-                  "on support for arrays with null elements.");
-            }
-            i += 1;
+          final int idx =
+              Arrays.asList(array.toArray(new Object[0])).indexOf(null);
```

Comment:
```suggestion
              Arrays.asList(array).indexOf(null);
```

Reply:
@Fokko I fell into this trap once myself.  What you are suggesting is not correct.  `array` here is a Collection.  Without first turning the Collection into a native array, you get an Array with a single element in it... the pointer (reference) to the Collection itself, not the contents of the Collection.